coda/coda_mdstore/tests/test_views.py

```diff
@@ -591,22 +591,42 @@ def test_raises_http404_file_handle_is_falsy(self, rf):
         with pytest.raises(http.Http404):
             views.bagURLList(request, bag.name)
 
-    @pytest.mark.xfail(reason='Refactor is required.')
-    def test_output_text(self):
-        assert 0
-
-    @pytest.mark.xfail(reason='Refactor is required.')
-    def test_top_files_block(self):
-        # TODO: rename once implemented.
-        assert 0
-
-    @pytest.mark.xfail(reason='Refactor is required.')
-    def test_path_is_not_unicode_safe(self):
-        assert 0
+    def test_output_text(self, rf):
+        """Test the outputText obtained from mocked getFileHandle object"""
+        self.getFileHandle.return_value.url = 'https://coda/testurl'
+        # Mock the return value of urllib.request.urlopen(url) in getFileHandle()
+        self.getFileHandle.return_value.readline.side_effect = [
+            '<html, class=coda lang=en dir=ltr>', '']
+        bag = FullBagFactory.create()
+        request = rf.get('/')
+        response = views.bagURLList(request, bag.name)
+        assert b'https://coda/class=coda lang=en dir=ltr>' in response.content
+        assert response.status_code == 200
 
-    @pytest.mark.xfail(reason='Refactor is required.')
-    def test_returns_status_code_200(self):
-        assert 0
+    def test_top_files_block(self, rf):
+        """Test getFileHandle and getFileList methods are called without any failures"""
+        self.getFileHandle.return_value.url = 'https://testurl'
+        # Mock the return value of urllib.request.urlopen(url) in getFileHandle()
+        self.getFileHandle.return_value.readline.side_effect = [
+            '<html, class=coda lang=en dir=ltr>', '']
+        bag = FullBagFactory.create()
+        request = rf.get('/')
+        text = """<html>
+                     <body>
+                     <tr> <td>test</td> <td>data</td> </tr>
+                     <tr> <td>of </td> </tr>
+                     <tr> <td>
+                        <a href='test.com'>url</a>
+                        <a href='testurl.com'>here</a>
+                     </td> </tr>
+                     </body>
+                  </html>"""
+        # Mock urlopen() in getFileHandle()
```

Comment:
I don't think this comment is needed as it doesn't add any clarification to the code--it is a pretty direct translation.

Reply:
How about  `Mock return value of getFileHandle()`? or do you think its not just necessary here?